RectangularTriangleService.rb

```diff
@@ -0,0 +1,64 @@
+class Triangle
+  class GigasIOError < StandardError; end
+
+  attr_accessor :side_a, :side_b, :side_c
+
+  def initialize(side_a, side_b, side_c)
+    @side_a = side_a
+    @side_b = side_b
+    @side_c = side_c
+  end
+
+  def call
+    input_check
+    calculate
+  rescue GigasIOError => e
+    warn e
+  end
+
+  private
+
+  def calculate
+    if pythagorean?
+      if isosceles?
```

Comment:
it is bad practise

Reply:
refactored the nested statement, hopefully its good to go now!